plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/containers/KeycloakContainer.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.containers;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.log.Logger;
+import io.trino.testing.containers.BaseTestContainer;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.testcontainers.containers.Network;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class KeycloakContainer
+        extends BaseTestContainer
+{
+    private static final Logger log = Logger.get(KeycloakContainer.class);
+
+    public static final String DEFAULT_IMAGE = "quay.io/keycloak/keycloak:21.1.1";
+    public static final String DEFAULT_HOST_NAME = "keycloak";
+
+    public static final String DEFAULT_USER_NAME = "admin";
+    public static final String DEFAULT_PASSWORD = "admin";
+
+    public static final int PORT = 8080;
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    private KeycloakContainer(String image, String hostName, Set<Integer> exposePorts, Map<String, String> filesToMount, Map<String, String> envVars, Optional<Network> network, int retryLimit)
+    {
+        super(image, hostName, exposePorts, filesToMount, envVars, network, retryLimit);
+    }
+
+    @Override
+    protected void setupContainer()
+    {
+        super.setupContainer();
+        withRunCommand(ImmutableList.of("start-dev"));
+    }
+
+    @Override
+    public void start()
+    {
+        super.start();
+        log.info("Keycloak container started with URL: %s", getUrl());
+    }
+
+    public String getUrl()
+    {
+        return "http://" + getMappedHostAndPortForExposedPort(PORT);
+    }
+
+    public static String getUrlWithHostName()
+    {
+        return "http://" + DEFAULT_HOST_NAME + ":" + PORT;
+    }
+
+    public String getAccessToken()
+    {
+        String realm = "master";
+        String clientId = "admin-cli";
+
+        try (Keycloak keycloak = KeycloakBuilder.builder()
+                .serverUrl(getUrl())
+                .realm(realm)
+                .clientId(clientId)
+                .username(DEFAULT_USER_NAME)
+                .password(DEFAULT_PASSWORD)
+                .build()) {
+            RealmResource master = keycloak.realms().realm(realm);
+            RealmRepresentation masterRep = master.toRepresentation();
+            // change access token lifespan from 1 minute (default) to 1 hour
```

Comment:
The comment explains what the code does, but not the reasoning behind it.

Reply:
added reasoning.